samples/net/coap_client/src/coap-client.c

```diff
@@ -211,9 +240,29 @@ static void event_iface_up(struct net_mgmt_event_callback *cb,
 
 	net_pkt_frag_add(pkt, frag);
 
-	r = coap_packet_init(&request, pkt, 1, COAP_TYPE_CON,
-			     8, coap_next_token(),
-			     COAP_METHOD_GET, coap_next_id());
+	switch(coap_method)
+	{
+		case 0:
```

Comment:
If we use type "enum coap_method" in the function interface, then we could use "COAP_METHOD__XXX" here as case id.